mopidy/core/library.py

```diff
@@ -117,12 +126,16 @@ def _browse(self, uri):
 
         with _backend_error_handling(backend):
             result = backend.library.browse(uri).get()
-            validation.check_instances(result, models.Ref)
+            validation.check_instances(result, Ref)
             return result
 
         return []
 
-    def get_distinct(self, field, query=None):
+    def get_distinct(
+        self,
+        field: DistinctField,
+        query: Optional[Query[DistinctField]] = None,
```

Comment:
Shouldn't this be `DistinctField`, as in everything in `SearchField` except `any`?

Reply:
If I remember correctly, it should include Any. I think I checked the mpd docs/source for this and made sure the doc string we had was accurate. I think it makes sense that the query supports Any. I can try and dig out any notes/work I did when looking at this.

Reply:
It was https://github.com/mopidy/mopidy/pull/1900 where I tried to align it all. To me it makes sense that the `query` should support `any`, that seems a reasonable query to make. But if you don't want to include `any` then you need to change the call to `validation.check_query` below so it's `validation.check_query(query, DISTINCT_FIELDS.keys())`

Reply:
Thinking just a few seconds more about this, it makes a lot of sense that the query that is only used for filtering is a full-blooded search query, while only the definition of what to return as a result requires specific field names instead of `any`. Good catch!